testsuites/listener/shared/client_sg/test_design_docs_update.py

```diff
@@ -82,3 +82,98 @@ def test_design_doc_update(setup_client_syncgateway_test):
+
+    log_info("Running 'test_design_doc_update'")
+
+    ls_url = setup_client_syncgateway_test["ls_url"]
+    log_info("ls_url: {}".format(ls_url))
+
+    client = MobileRestClient()
+
+    ls_url = setup_client_syncgateway_test["ls_url"]
+
+    num_content_docs_per_db = 10
+    num_no_content_docs_per_db = 5
+    d_doc_name = "dd"
+    ls_db = client.create_database(ls_url, name="ls_db")
+
+    # Add 10 docs to the client with content
+    bulk_docs_content = create_docs("doc_content_", num_content_docs_per_db, content={"hi": "I should be in the view"})
+    ls_db_docs1 = client.add_bulk_docs(url=ls_url, db=ls_db, docs=bulk_docs_content)
+    assert len(ls_db_docs1) == num_content_docs_per_db
+
+    # Add 5 docs to the client without content
+    bulk_docs_no_content = create_docs("doc_no_content_", num_no_content_docs_per_db)
+    ls_db_docs2 = client.add_bulk_docs(url=ls_url, db=ls_db, docs=bulk_docs_no_content)
+    assert len(ls_db_docs2) == num_no_content_docs_per_db
+
+    # Design doc to to fetch doc._id, doc._rev for docs with content
+    view = """{
+    "language" : "javascript",
+    "views" : {
+        "content_view" : {
+            "map" : "function(doc, meta) { if (doc.content) { emit(doc._id, doc._rev); } }"
+        }
+    }
+}"""
+
+    # querying docs with newly created design doc
+    client.add_design_doc(url=ls_url, db=ls_db, name=d_doc_name, doc=view)
+    content_view_rows_1 = client.get_view(url=ls_url, db=ls_db, design_doc_name=d_doc_name, view_name="content_view")
+    client.verify_view_row_num(view_response=content_view_rows_1, expected_num_rows=10)
+
+    # deleting the design doc
+    dd_rev = client.get_design_doc_rev(url=ls_url, db=ls_db, name=d_doc_name)
+    assert dd_rev and dd_rev.startswith("1-")
+    log_info("dd_rev: {}".format(dd_rev))
+    log_info("Deleting design doc - dd")
+    client.delete_design_doc(url=ls_url, db=ls_db, name=d_doc_name, rev_id=dd_rev)
+
+    # verifying that we can't view query for deleted design doc
+    try:
+        client.get_view(url=ls_url, db=ls_db, design_doc_name=d_doc_name, view_name="content_view")
+        assert False, "Deleting the design doc doesn't also deleting the view index."
+    except Exception, err:
+        err_msg = "404 Client Error"
+        assert err_msg in str(err)
+
+    # Design doc to to fetch doc._id, doc._rev for docs with no content
+    view = """{
+    "language" : "javascript",
+    "views" : {
+        "content_view" : {
+            "map" : "function(doc, meta) { if (!(doc.content)) { emit(doc._id, doc._rev); } }"
+        }
+    }
+}"""
+
+    # creating a new Desing doc  with same name as previous design doc
+    log_info("Creating a new design doc to query docs with no content with same name as previous one")
+    client.add_design_doc(url=ls_url, db=ls_db, name=d_doc_name, doc=view)
+    dd_rev_new = client.get_design_doc_rev(url=ls_url, db=ls_db, name=d_doc_name)
+#     assert dd_rev_new != dd_rev and dd_rev_new.startswith("1-")
```

Comment:
can you remove this comment if it is not necessary

Reply:
removed.